src/fabric/src/fabric2_fdb.erl

```diff
@@ -1222,12 +1245,14 @@ execute_transaction(Tx, Fun, LayerPrefix) ->
             erlfdb:set(Tx, get_transaction_id(Tx, LayerPrefix), <<>>),
             put(?PDICT_TX_RES_KEY, Result)
     end,
+    ok = run_on_commit_fun(Tx),
     Result.
 
 
 clear_transaction() ->
     fabric2_txids:remove(get(?PDICT_TX_ID_KEY)),
     erase(?PDICT_CHECKED_DB_IS_CURRENT),
+    erase(?PDICT_CHECKED_MD_IS_CURRENT),
```

Comment:
We should probably clear `PDICT_ON_COMMIT_FUN` as well right?

Reply:
We clear it after we run the function successfully. But we'd ahve to clear it even if we fail. Good point

Reply:
Also good call on moving it above since `Tx` is part of the key.